tests/acceptance/features/bootstrap/GraphApiContext.php

```diff
@@ -0,0 +1,211 @@
+     */
+    public function sendCreateSpaceRequest(
+        $baseUrl,
+        $user,
+        $password,
+        string $spaceName,
+        string $xRequestId = '',
+        array $headers = []
+    ): ResponseInterface
+    {
+        $fullUrl = $baseUrl;
+        if (!str_ends_with($fullUrl, '/')) {
+            $fullUrl .= '/';
+        }
+        $fullUrl .= "drives/" . $spaceName;
+
+        return HttpRequestHelper::sendRequest($fullUrl, $xRequestId, 'POST', $user, $password, $headers);
+    }
+
+    /**
+     * Send Propfind Request to Url
+     *
+     * @param $fullUrl
+     * @param $user
+     * @param $password
+     * @param string $xRequestId
+     * @param array $headers
+     * @return ResponseInterface
+     */
+    public function sendPropfindRequestToUrl(
+        $fullUrl,
+        $user,
+        $password,
+        string $xRequestId = '',
+        array $headers = []
+    ): ResponseInterface
+    {
+        return HttpRequestHelper::sendRequest($fullUrl, $xRequestId, 'PROPFIND', $user, $password, $headers);
+    }
+
+    /**
+     * @When /^user "([^"]*)" lists all available spaces via the GraphApi$/
+     *
+     * @param $user
+     * @return void
+     */
+    public function theUserListsAllHisAvailableSpacesUsingTheGraphApi($user): void
+    {
+        $this->featureContext->setResponse(
+            $this->listSpacesRequest(
+                $this->featureContext->getBaseUrl(),
+                $user,
+                $this->featureContext->getPasswordForUser($user),
+                "",
+                ""
+            )
+        );
+    }
+
+    /**
+     * @Then the webDavUrl of the personal space has been found
+     *
+     * @return void
+     */
+    public function theWebDavUrlOfThePersonalSpaceHasBeenFound(): void
+    {
+        $rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
+        $drives = [];
+        if (isset(\json_decode($rawBody,  true)["value"])) {
+            $drives = \json_decode($rawBody, true)["value"];
+        }
+
+        Assert::assertArrayHasKey(0, $drives, "No drives were found on that endpoint");
+
+        foreach($drives as $drive) {
+            if (isset($drive["driveType"]) && $drive["driveType"] === "personal") {
+                $this->setPersonalDriveWebDavUrl($drive["root"]["webDavUrl"]);
+
+                Assert::assertNotEmpty(
+                    $drive["root"]["webDavUrl"],
+                    "The personal space attributes contain no webDavUrl"
+                );
+            }
+        }
+    }
+
+    /**
+     * @When /^user "([^"]*)" lists the content of the personal space root using the WebDav Api$/
+     *
+     * @param $user
+     *
+     * @return void
+     */
+    public function theUserListsTheContentOfASpaceRootUsingTheWebDAvApi($user): void
+    {
+        $this->featureContext->setResponse(
+            $this->sendPropfindRequestToUrl(
+                $this->getPersonalDriveWebDavUrl(),
```

Comment:
This `When` step depends on `Then the webDavUrl of the personal space has been found` because that is the only place where `personalDriveWebDavUrl` is set
I think a `When` step should be independent of any `Then` step, otherwise it cannot be used in a normal `Given - When - Then` order
The easy solution would be to call `theUserListsAllHisAvailableSpacesUsingTheGraphApi()` and `theWebDavUrlOfThePersonalSpaceHasBeenFound()` from inside this function